controllers/todos.js

```diff
@@ -0,0 +1,111 @@
+const { Todo } = require ('../models')
+
+
+class TodoController {
+    static createTodo (req, res, next) {
+        const {title, description, status, due_date} = req.body
+        Todo   
+            .create({
+                title,
+                description,
+                status,
+                due_date : new Date (due_date)
+            })
+            .then(todo => {
+                res.status(201).json({todo})
+            })
+            .catch(err => {
+                next(err)
+            })
+    }
+
+    static findAll (req, res){
+        Todo
+            .findAll()
+            .then(todos => {
+                res.status(200).json(todos)
+            })
+            .catch (err => {
+                res.send(err)
+            })
+    }
+
+    static findById (req, res, next) {
+        const id = req.params.id
+        Todo
+            .findByPk(id)
+            .then(todo => {
+               if (!todo){
+                   res.status(404).json({error : `not found`})
+               }
+               res.status(200).json({todo})
+            })
+            .catch ( err => {
+                next(err)
+            })
+    }
+
+    static editTodo (req, res, next) {
+        const {title, description, status, due_date} = req.body
+        const idParams = req.params.id
+        Todo    
+            .update ({
+                title,
+                description,
+                status,
+                due_date : new Date (due_date)
+            },{
+                where : { id : idParams},
+                returning : true
+            })
+            .then ( todo => {
+                if (!todo){
+                    res.status(404).json({error : `not found`})
+                }
+
+                res.status(200).json(todo[1][0])
+            })
+            .catch (err => {
+                next(err)
+            })
+    }
+
+    static changeStatusTodo (req, res) {
+        const {status} = req.body
+        const idParams = req.params.id
+        Todo
+            .update({
+                status : status
+            }, {
+                where : {id : idParams},
+                returning: true
+            })
+            .then(todo => {
+                if (!todo){
```

Comment:
ini musti di cek data[0] nya angkanya berapa kalo 0 berarti not found

Reply:
maksudnya hasil return dari updatenya itu mas? kan bentuknya array yang isinya angka tuh ya? kalo 0 berati not found, kalo ada berati di tunjukin hasil updatenya. gitu kah mas?